### PR TITLE
Fix cart2sph to reflect correct range and update tests.

### PR DESCRIPTION
**Pull request by chris-hld**
This is the PR for https://github.com/sfstoolbox/sfs-python/issues/59.
Testing against `np.arctan2` is not meaningful, since this is causing the range issue.

### CODE REVIEW

**Comment by mgeier** [comment]
Thanks for this PR and sorry for the late response!

As mentioned in https://github.com/sfstoolbox/sfs-python/issues/59#issuecomment-388902384, this destroys a few of the "special" values mentioned in https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.arctan2.html.

For example, those are (first result is the previous behavior, second result is this PR):

```python
import numpy as np
import sfs

sfs.util.cart2sph(0, -0., 1)[0]
# -0.0
# 0.0
sfs.util.cart2sph(np.inf, -0., 0)[0]
# -0.0
# 0.0
sfs.util.cart2sph(-0., -0., 1)[0]
# -3.141592653589793
# 3.141592653589793
sfs.util.cart2sph(-np.inf, -0., 0)[0]
# -3.141592653589793
# 3.141592653589793
```

I think from a geometrical point of view, the old values make a lot of sense, don't they?

**Comment by chris-hld** [comment]
Ok, now I see your point more clearly. 

Here is my opinion on this:
We state that the azimuth is in `[0, 2pi)` . This is getting a bit philosophically speaking, but this interval does not include `-0`. So, if any of this, `(0, -0., 1)` may return `2pi`, which is not included in the domain, though.
For the last two examples I'd say generally the second value is as correct as the first. Again, our domain is `[0, 2pi)`, so the limit is `pi`, not `-pi`, the latter is simply not approachable. The explanation is that we are approaching pi from the positive direction 2pi. 
This is just different than atan2, which approaches this value continuously from `-0`, because it has a different domain. To my knowledge this is actually not the same, neither is this a special value.
So in short:
atan2 does `-0 -> -pi/2 -> -pi`.
We should do `2pi -> 3/2 pi -> pi`.
atan2 is discontinuous in `pi/-pi`, we are in `0/2pi`.
Both values are correct, but not the same.

Honestly, these are special values that we never actually use or treat separately, are we?
Are we catching these special values somewhere?
Otherwise it's pretty confusing if we say our values are within a certain range in the docs but we actually don't provide them. However, I agree that it shouldn't make a difference if any following implementation is correct.

**Comment by mgeier** [comment]
> We state that the azimuth is in `[0, 2pi)`

Yes we do, but we probably shouldn't (as discussed in #59).

As mentioned in #59, any function that doesn't work with the full input range of real numbers is broken.
So for all practical use cases we have considered so far, it doesn't matter one bit what range we return.

But of course we have to choose some range.
Above I've given some motivation why the range `[-pi, pi]` makes sense.
Incidentally, that also leads to the simplest implementation.
I don't see a reason why we should jump through hoops in our implementation for zero gain.

In a theoretical math context another range may or may not make more sense, but I don't see a problem in using two different ranges for theoretical and practical work.

Of course, it should be documented correctly.

> For the last two examples I'd say generally the second value is as correct as the first.

That may be the case for some definitions of "correct", but the second value definitely throws away information compared to the first value.

> Again, our domain is `[0, 2pi)`

Again, it doesn't have to be.

>  To my knowledge this is actually not the same, neither is this a special value.

I don't understand this sentence, can you please clarify?

> Honestly, these are special values that we never actually use or treat separately, are we?

Probably not. But I don't see a reason why we should throw away this information.

> Are we catching these special values somewhere?

I guess not. At least not yet.

But since we are implementing a library here, we don't have all possible use cases available to us.

In this case I think we should try to do The Right Thing.

> Otherwise it's pretty confusing if we say our values are within a certain range in the docs but we actually don't provide them.

I agree. That's why we should change the documentation, see https://github.com/sfstoolbox/sfs-python/issues/59#issuecomment-386634275.

Unless somebody provides a real reason to change the behavior.

**Comment by chris-hld** [comment]
> But since we are implementing a library here, we don't have all possible use cases available to us.
>In this case I think we should try to do The Right Thing.

I agree. 
So then, let's only update the docs. I personally think it is still better to test against pre-calculated values, and not the return of `np.arctan2()` (as the current test does). What do you think?

**Comment by chris-hld** [comment]
Closing and moving back to issue #59.

**Comment by mgeier** [comment]
> I personally think it is still better to test against pre-calculated values, and not the return of `np.arctan2()` (as the current test does). What do you think?

I agree.

Would you mind extracting your changes to the tests from this PR and making a new PR only for the tests?
